src/App.jsx

```diff
@@ -1,28 +1,88 @@
 import React from 'react';
+
 import './App.scss';
 import './styles/general.scss';
+
+import { getTodos } from './components/api/api';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
 
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
+    searchTitle: '',
+    searchTodoStatus: '',
   };
 
+  async componentDidMount() {
+    const todos = await getTodos();
+
+    this.setState({
+      todos: todos.data,
+    });
+  }
+
+  updateSearchTitle = (value) => {
+    this.setState({ searchTitle: value });
+  };
+
+  updateSearchTodoStatus = (status) => {
+    this.setState({ searchTodoStatus: status });
+  }
+
+  visibleTodos = (todos) => {
+    const { searchTitle, searchTodoStatus } = this.state;
+
+    const normalizeQuery = searchTitle.toLowerCase().trim();
+
+    let visibleTodos = [...todos];
```

Comment:
```suggestion
    let visibleTodos;
```
You may not create here the copy of the initial array.
Later in your code, you may filter directly the `todos` array
`visibleTodos = todos.filter....`